CMakeLists.txt

```diff
@@ -316,8 +316,8 @@ OCV_OPTION(INSTALL_TESTS            "Install accuracy and performance test binar
 
 # OpenCV build options
 # ===================================================
-OCV_OPTION(ENABLE_TYPE_SAFE_API       "Enable enum-based type-safe API? If disabled, int types will be utilized" ON)
-OCV_OPTION(ENABLE_COMPATIBLE_API      "Enable backwards compatibility with legacy API?"          ON IF ENABLE_TYPE_SAFE_API)
+OCV_OPTION(DISABLE_TYPE_SAFE_API      "Disable enum-based type-safe API? If disabled, 'int' types will be utilized" OFF)
+OCV_OPTION(DISABLE_TYPE_COMPATIBLE_API "Enable backwards compatibility with legacy API?"         ON IF DISABLE_TYPE_SAFE_API)
```

Comment:
Should not this be `DISABLE_TYPE_COMPATIBLE_API "..." OFF if NOT DISABLE_TYPE_SAFE_API`?